cmd/dogstatsd/subcommands/start/command.go

```diff
@@ -145,13 +142,14 @@ func RunDogstatsdFct(cliParams *CLIParams, defaultConfPath string, defaultLogFil
 	)
 }
 
-func start(cliParams *CLIParams, config config.Component, log log.Component, params *Params, server dogstatsdServer.Component, sharedForwarder defaultforwarder.Component, wmeta workloadmeta.Component, demultiplexer demultiplexer.Component, metadataRunner runner.Component, hostComp host.Component) error { //nolint:revive // TODO fix revive unusued-parameter
+func start(cliParams *CLIParams, config config.Component, log log.Component, params *Params, server dogstatsdServer.Component, sharedForwarder defaultforwarder.Component, wmeta optional.Option[workloadmeta.Component], demultiplexer demultiplexer.Component, metadataRunner runner.Component, hostComp host.Component) error { //nolint:revive // TODO fix revive unusued-parameter
 	// Main context passed to components
 	ctx, cancel := context.WithCancel(context.Background())
 
+	w, _ := wmeta.Get()
 	components := &DogstatsdComponents{
 		DogstatsdServer: server,
-		WorkloadMeta:    wmeta,
+		WorkloadMeta:    w,
```

Comment:
Ideally `WorkloadMeta` type could be `optional.Option[workloadmeta.Component]`.

Reply:
To avoid running the CI again and hold back the train after the freeze, I'll do this in a follow-up. But that makes sense, I can make that change.